public/app/features/annotations/standardAnnotationSupport.ts

```diff
@@ -231,13 +231,21 @@ export function getAnnotationsFromData(
 // annotation support API needs some work to support less "standard" editors like prometheus and here it is not
 // polluting public API.
 
+const legacyRunner = [
+  'prometheus',
+  'loki',
+  'elasticsearch',
+  'grafana-opensearch-datasource', // external
+];
+
 /**
  * Opt out of using the default mapping functionality on frontend.
  */
 export function shouldUseMappingUI(datasource: DataSourceApi): boolean {
   const { type } = datasource;
-  return (
-    type !== 'prometheus' && type !== 'elasticsearch' && type !== 'loki' && type !== 'grafana-opensearch-datasource'
+  return !(
+    type === 'datasource' || //  ODD behavior for "-- Grafana --" datasource
```

Comment:
@ashharrison90 so now we know why this was [here](https://github.com/grafana/grafana/pull/48814/files#diff-1c32b50f4e6010b94cfb26f3c99f93a48a0afae330294d206a9eeaa9393b53abL181) before.